bofire/utils/torch_tools.py

```diff
@@ -182,24 +184,32 @@ def min_constraint(indices: Tensor, num_features: int, min_count: int):
         )
         if c.max_count != len(c.features):
             constraints.append(
-                max_constraint(
-                    indices=indices,
-                    num_features=len(c.features),
-                    max_count=c.max_count,
-                ),
+                (
+                    max_constraint(
+                        indices=indices,
+                        num_features=len(c.features),
+                        max_count=c.max_count,
+                    ),
+                    True,
+                )
             )
         if c.min_count > 0:
             constraints.append(
-                min_constraint(
-                    indices=indices,
-                    num_features=len(c.features),
-                    min_count=c.min_count,
-                ),
+                (
+                    min_constraint(
+                        indices=indices,
+                        num_features=len(c.features),
+                        min_count=c.min_count,
+                    ),
+                    True,
+                )
             )
     return constraints
 
 
-def get_product_constraints(domain: Domain) -> List[Callable[[Tensor], float]]:
+def get_product_constraints(
+    domain: Domain,
+) -> List[Tuple[Callable[[Tensor], float], bool]]:
```

Comment:
The longish type `Tuple[Callable[[Tensor], float], bool` is somehow difficult to understand. Wouldn't it make sense to use a type like
```python
class Constraint(BaseModel):
    evaluate: Callable[[Tensor], float]
    is_intrapoint: bool
```
and then use `list[Constraint]` as return type?

Reply:
In principle yes, but I would like to keep it as is, becuase this is exactly the format which is required by the botorch api. Else, we have an additional conversion step. If ok for you, I would merge the current version of the PR.

Reply:
Could you add the role of the bool to the docstring?